upload-notebooks/upload-notebook.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+# Upload notebook to TileDB Cloud
+#
+# Usage:
+#     python upload-notebook.py <path/to/notebook.ipynb>
+#     python upload-notebook.py --help
+#
+# Variables read from the environment:
+#   - TILEDB_CLOUD_TOKEN: API token for TileDB Cloud account (required)
+#   - TILEDB_CLOUD_NAMESPACE: Namespace for TileDB Cloud (required)
+#   - TILEDB_CLOUD_STORAGE_PATH: Storage path on TileDB Cloud (optional)
+#   - TILEDB_CLOUD_STORAGE_CREDENTIAL_NAME: Storage credentials to use on TileDB Cloud (optional)
+
+
+import argparse
+import os
+import sys
+
+import tiledb.cloud
+
+# Process input
+parser = argparse.ArgumentParser(
+    description="Upload notebook to TileDB Cloud",
+    epilog="TileDB Cloud options may also be set as environment variables, but values set via command line flags will take priority",
+)
+parser.add_argument(
+    "notebook",
+    help="Notebook file to upload (extension .ipynb)",
+)
+parser.add_argument(
+    "--tiledb-cloud-token",
+    required=False,
+    default=os.environ.get("TILEDB_CLOUD_TOKEN"),
+    help="API token for TileDB Cloud account",
+)
+parser.add_argument(
+    "--tiledb-cloud-namespace",
+    required=False,
+    default=os.environ.get("TILEDB_CLOUD_NAMESPACE"),
+    help="Namespace for TileDB Cloud",
+)
+parser.add_argument(
+    "--tiledb-cloud-storage-path",
+    required=False,
+    default=os.environ.get("TILEDB_CLOUD_STORAGE_PATH"),
+    help="Storage path on TileDB Cloud",
+)
+parser.add_argument(
+    "--tiledb-cloud-storage-credential-name",
+    required=False,
+    default=os.environ.get("TILEDB_CLOUD_STORAGE_CREDENTIAL_NAME"),
+    help="Storage credentials to use on TileDB Cloud",
+)
+args = parser.parse_args()
+
+notebook = args.notebook
+token = args.tiledb_cloud_token
+namespace = args.tiledb_cloud_namespace
+storage_path = args.tiledb_cloud_storage_path
+storage_credential_name = args.tiledb_cloud_storage_credential_name
+
+# Verify input
+if not os.path.exists(notebook):
+    sys.stderr.write("Error: Notebook file does not exist: %s\n" % (notebook))
```

Comment:
you can use `print(..., file=sys.stderr)`. Also using a f-string might make this more readable: `"Error: notebook file {notebook!r} does not exist"`